src/main/java/com/wavefront/sdk/entities/histograms/WavefrontHistogramImpl.java

```diff
@@ -42,45 +40,26 @@
   private final Supplier<Long> clockMillis;
 
   /**
-   * Global concurrent list of thread local histogramBinsList wrapped in WeakReference.
-   * This list holds all the thread local List of Minute Bins.
+   * Global concurrent list of ThreadMinuteBin.
    * This is ConcurrentLinkedDeque so that we can support 'flatMap(List::stream)' without
    * worrying about ConcurrentModificationException.
-   * The MinuteBin itself is not thread safe and can change but it is still thread safe since we
-   * don’t ever update a bin that’s old or flush a bin that’s within the current minute.
    */
-  private final List<WeakReference<ConcurrentLinkedDeque<MinuteBin>>> globalHistogramBinsList =
-      new ArrayList<>();
-
-  private final StampedLock stampedLock = new StampedLock();
-  // Protects read access to globalHistogramBinsList
-  private final Lock readLock = stampedLock.asReadLock();
-
-  // Protects write access to globalHistogramBinsList
-  private final Lock writeLock = stampedLock.asWriteLock();
+  private final ConcurrentLinkedDeque<ThreadMinuteBin> globalHistogramBinsList = new ConcurrentLinkedDeque<>();
 
   /**
-   * ThreadLocal histogramBinsList where the initial value set is also added to a
-   * global list of thread local histogramBinsList wrapped in WeakReference
+   * Current Minute Histogram Bin.
+   * Update functions will only update data inside currentMinuteBin, which contains
+   * Timestamp and the ConcurrentMap of ThreadId and TDigest distribution.
    */
-  private final ThreadLocal<ConcurrentLinkedDeque<MinuteBin>> histogramBinsList =
-      ThreadLocal.withInitial(() -> {
-        ConcurrentLinkedDeque<MinuteBin> sharedBinsInstance = new ConcurrentLinkedDeque<>();
-        try {
-          writeLock.lock();
-          globalHistogramBinsList.add(new WeakReference<>(sharedBinsInstance));
-        } finally {
-          writeLock.unlock();
-        }
-        return sharedBinsInstance;
-  });
+  private ThreadMinuteBin currentMinuteBin;
```

Comment:
Can you add a comment as to why this is not final and how updating this reference is done in thread safe manner below.